processing/threedidepth_algorithm.py

```diff
@@ -0,0 +1,293 @@
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+from collections import namedtuple
+import logging
+import os
+
+from processing.gui.NumberInputPanel import NumberInputPanel
+from processing.gui.wrappers import WidgetWrapper, DIALOG_STANDARD
+from qgis.PyQt import uic
+from qgis.PyQt.QtCore import QCoreApplication
+from qgis.core import (
+    QgsFeedback,
+    QgsProcessingAlgorithm,
+    QgsProcessingParameterEnum,
+    QgsProcessingParameterFile,
+    QgsProcessingParameterNumber,
+    QgsProcessingParameterRasterDestination,
+    QgsProcessingParameterRasterLayer,
+)
+
+import h5py
+from threedidepth.calculate import calculate_waterdepth
+from threedidepth.calculate import (
+    MODE_LINEAR,
+    MODE_CONSTANT,
+    MODE_LINEAR_S1,
+    MODE_CONSTANT_S1,
+    MODE_LIZARD,
+    MODE_LIZARD_S1
+)
+
+
+logger = logging.getLogger(__name__)
+pluginPath = os.path.split(os.path.dirname(__file__))[0]
+Mode = namedtuple("Mode", ["name", "description"])
+
+
+class ProcessingParamterNetcdfNumber(QgsProcessingParameterNumber):
+    def __init__(self, *args, parentParameterName='', **kwargs):
+        super().__init__(*args, **kwargs)
+        self.parentParameterName = parentParameterName
+        self.setMetadata(
+            {'widget_wrapper': {'class': ThreediResultTimeSliderWidget}}
+        )
+
+
+WIDGET, BASE = uic.loadUiType(os.path.join(
+    pluginPath, 'processing', 'ui', 'widgetTimeSlider.ui')
+)
+
+
+class TimeSliderWidget(BASE, WIDGET):
+    """Timeslider form widget
+
+    Provide a horizontal slider and an LCD display connected to the slider.
+    """
+
+    def __init__(self):
+        super(TimeSliderWidget, self).__init__(None)
+        self.setupUi(self)
+        self.horizontalSlider.valueChanged.connect(self.set_lcd_value)
+        self.index = None
+        self.timestamps = None
+        self.reset()
+
+    def getValue(self):
+        return self.index
+
+    def set_timestamps(self, timestamps):
+        self.setDisabled(False)
+        self.horizontalSlider.setMinimum(0)
+        self.horizontalSlider.setMaximum(len(timestamps) - 1)
+        self.timestamps = timestamps
+
+    def set_lcd_value(self, index: int):
+        self.index = index
+        if self.timestamps is not None:
+            value = self.timestamps[index]
+        else:
+            value = 0
+        lcd_value = self.format_lcd_value(value)
+        self.lcdNumber.display(lcd_value)
+
+    def format_lcd_value(self, value: float) -> str:
+        days = int(value // 86400)
+        hours = int((value // 3600) % 24)
+        minutes = int((value // 60) % 60)
```

Comment:
the `divmod` function is a good way to do things like. It returns both division result and remainder in a tuple (division, remainder):
Example:
```
weeks, seconds = divmod(seconds, 7*24*60*60)
days, seconds = divmod(seconds, 24*60*60)
hours, seconds = divmod(seconds, 60*60)
minutes, seconds = divmod(seconds, 60)
```